### PR TITLE
Integrate barf (prefetchers)

### PR DESCRIPTION
**Pull request by jerryz123**
BARF (BAR-fetchers) is a generic API for describing simple prefetchers, that can be instantiated in front of L1 HellaCaches, or in TileLink bus widgets.

Currently, a handful of example algorithms are implemented.

**Related PRs / Issues**:

**Type of change**:
- [ ] Bug fix
- [x] New feature
- [ ] Other enhancement

**Impact**:
- [ ] RTL change
- [ ] Software change (RISC-V software)
- [ ] Build system change
- [x] Other

**Contributor Checklist**:
- [x] Did you set `main` as the base branch?
- [x] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [x] Did you state the type-of-change/impact?
- [x] Did you delete any extraneous prints/debugging code?
- [x] Did you mark the PR with a `changelog:` label?
- [ ] (If applicable) Did you update the conda `.conda-lock.yml` file if you updated the conda requirements file?
- [ ] (If applicable) Did you add documentation for the feature?
- [ ] (If applicable) Did you add a test demonstrating the PR?

- [ ] (If applicable) Did you mark the PR as `Please Backport`?

### CODE REVIEW

File: .github/scripts/run-tests.sh [hunk 1]
```diff
@@ -56,6 +56,9 @@ case $1 in
     chipyard-hetero)
         run_bmark ${mapping[$1]}
         ;;
+    chipyard-prefetchers)
+        make -C $LOCAL_SIM_DIR $DISABLE_SIM_PREREQ ${mapping[$1]} run-binary BINARY=$RISCV/riscv64-unknown-elf/share/riscv-tests/benchmarks/dhrystone.riscv
```
Comment (abejgonzalez) [inline .github/scripts/run-tests.sh:60]: Side note: Why don't we make LOADMEM the default? Isn't it fairly robust now?

**Review by abejgonzalez — CHANGES_REQUESTED** [review]

File: .github/scripts/run-tests.sh (hunk 1 above)
Comment (jerryz123) [reply .github/scripts/run-tests.sh:60]: LOADMEM only works on configs that rely on the normal backing memory model. For configs which rely on on-chip memory, or which have a more sophisticated memory system, setting LOADMEM may result in a silent failure that is difficult to debug.

File: generators/chipyard/src/main/scala/config/RocketConfigs.scala [hunk 2]
```diff
@@ -126,3 +126,11 @@ class CustomIOChipTopRocketConfig extends Config(
   new chipyard.example.WithCustomIOCells ++
   new freechips.rocketchip.subsystem.WithNBigCores(1) ++         // single rocket-core
   new chipyard.config.AbstractConfig)
+
+class PrefetchingRocketConfig extends Config(
+  new barf.WithHellaCachePrefetcher(Seq(0), barf.SingleStridedPrefetcherParams()) ++   // strided prefetcher, sits in front of the L1D$, monitors core requests to prefetching into the L1D$
+  new barf.WithTLICachePrefetcher(barf.MultiNextLinePrefetcherParams()) ++             // next-line prefetcher, sits between L1I$ and L2, monitors L1I$ misses to prefetch into L2
+  new barf.WithTLDCachePrefetcher(barf.SingleAMPMPrefetcherParams()) ++                // AMPM prefetcher, site between L1D$ and L2, monitors L1D$ misses to prefetch into L2
```
Comment (abejgonzalez) [inline generators/chipyard/src/main/scala/config/RocketConfigs.scala:133]:
```suggestion
  new barf.WithTLDCachePrefetcher(barf.SingleAMPMPrefetcherParams()) ++                // AMPM prefetcher, sits between L1D$ and L2, monitors L1D$ misses to prefetch into L2
```

**Review by abejgonzalez — APPROVED** [review]
LGTM minus the failing test.

**Pull request merged**